### PR TITLE
Sk ez pp/hist data load

### PR DESCRIPTION
**Pull request by gsa-suk**
https://github.com/GSA-TTS/FAC/issues/2718

census_to_gsafac application containing two management commands 
1. fac_s3 command to load census csv files from a local data folder to fac-census-to-gsafac-s3 bucket.
2. csv_to_postgres command to load data from csv files in fac-census-to-gsafac-s3/data bucket into postgres database.

README file contains details about how to run the commands.

Upon completion of the commands, the following messages will be displayed in the console:
Screenshot 2023-11-07 at 1 11 30 PM

## PR checklist: submitters

- [ ]   Link to an issue if possible. If there’s no issue, describe what your branch does. Even if there is an issue, a brief description in the PR is still useful.
- [ ]   List any special steps reviewers have to follow to test the PR. For example, adding a local environment variable, creating a local test file, etc.
- [ ]   For extra credit, submit a screen recording like [this one](https://github.com/GSA-TTS/FAC/pull/1821).
- [ ]   Make sure you’ve merged `main` into your branch shortly before creating the PR. (You should also be merging `main` into your branch regularly during development.)
- [ ]   Make sure you’ve accounted for any migrations. When you’re about to create the PR, bring up the application locally and then run `git status | grep migrations`. If there are any results, you probably need to add them to the branch for the PR. Your PR should have only **one** new migration file for each of the component apps, except in rare circumstances; you may need to delete some and re-run `python manage.py makemigrations` to reduce the number to one. (Also, unless in exceptional circumstances, your PR should not delete any migration files.)
- [ ]   Make sure that whatever feature you’re adding has tests that cover the feature. This includes test coverage to make sure that the previous workflow still works, if applicable.
- [ ]   Make sure the full-submission.cy.js [Cypress test](https://github.com/GSA-TTS/FAC/blob/main/docs/testing.md#end-to-end-testing) passes, if applicable.
- [ ]   Do manual testing locally. Our tests are not good enough yet to allow us to skip this step. If that’s not applicable for some reason, check this box.
- [ ]   Verify that no Git surgery was necessary, or, if it was necessary at any point, repeat the testing after it’s finished.
- [ ]   Once a PR is merged, keep an eye on it until it’s deployed to dev, and do enough testing on dev to verify that it deployed successfully, the feature works as expected, and the happy path for the broad feature area (such as submission) still works.

## PR checklist: reviewers

- [ ]   Pull the branch to your local environment and run `make docker-clean; make docker-first-run && docker compose up`; then run `docker compose exec web /bin/bash -c "python manage.py test"`
- [ ]   Manually test out the changes locally, or check this box to verify that it wasn’t applicable in this case.
- [ ]   Check that the PR has appropriate tests. Look out for changes in HTML/JS/JSON Schema logic that may need to be captured in Python tests even though the logic isn’t in Python.
- [ ]   Verify that no Git surgery is necessary at any point (such as during a merge party), or, if it was, repeat the testing after it’s finished.

The larger the PR, the stricter we should be about these points.

### CODE REVIEW

**Comment by sambodeme** [comment]
As mentioned in [here](https://docs.google.com/presentation/d/1jSulTDXuQckF-4qldACBvCyzBrhFyzISR1mOwRFgMmE/edit#slide=id.g295bfd633d4_0_41),  the keyword `c2g` should be changed to something like `census_to_gsafac` everywhere in this PR

**Comment by phildominguez-gsa** [comment]
To fix the linting error, append `|census_historical_migration/migrations` to the end of `exclude = ".venv/|audit/migrations/|dissemination/migrations"` in `pyproject.toml` (once we've moved everything into a single app).

**Comment by sambodeme** [comment]
> Do we need a new app for this `census_to_gsafac`? Question for @jadudm / @danswick. Since we already have `census_historical_migration` app in the tree, we could just add this code to it.

We decided to consolidate `census_to_gsafac` and `census_historical_migration` into a single app (`census_historical_migration`)

File: backend/census_historical_migration/management/commands/csv_to_postgres.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        Populate Postgres database from csv files
+        Usage:
+        manage.py csv_to_postgres --folder <folder_name> --clean <True|False>
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument("--folder", help="S3 folder name")
+        parser.add_argument("--clean")
+        parser.add_argument("--sample")
+        parser.add_argument("--load")
```
Comment (phildominguez-gsa) [inline backend/census_historical_migration/management/commands/csv_to_postgres.py:39]: Would be helpful to flesh these options out by including `help`/`type`/`required`/`default` params

**Review by JeanMarie-TTS — CHANGES_REQUESTED** [review]
fac_s3 ishould be part of `support` app. Including it here will result in duplication when the other PR related to backups gets merged. We can add it to the support app as part of this PR. The duplication will be avoided when the other PR is merged.
Please clarify why you have added more exclusions in pyproject.toml?.

**Comment by gsa-suk** [comment]
> fac_s3 ishould be part of `support` app. Including it here will result in duplication when the other PR related to backups gets merged. We can add it to the support app as part of this PR. The duplication will be avoided when the other PR is merged. Please clarify why you have added more exclusions in pyproject.toml?.

**Comment by sambodeme** [comment]
> fac_s3 ishould be part of `support` app. Including it here will result in duplication when the other PR related to backups gets merged. We can add it to the support app as part of this PR. The duplication will be avoided when the other PR is merged. Please clarify why you have added more exclusions in pyproject.toml?.

We exclude migration files from mypy checks in pyproject.toml to prevent this [linting issue](https://github.com/GSA-TTS/FAC/actions/runs/6778272321/job/18423469409). For context, mypy flags Django migration files with the error `error: Need type annotation for "dependencies" (hint: "dependencies: List[<type>] = ...") [var-annotated]`. Since these files are auto-generated by Django, we instruct mypy to exclude them from its linting process

**Review by JeanMarie-TTS — APPROVED** [review]

**Review by sambodeme — APPROVED** [review]

**Pull request merged**